libkbfs/folder_branch_ops.go

```diff
@@ -3789,10 +3743,85 @@ func (fbo *folderBranchOps) syncAllLocked(
 			fbo.dirOps = nil
 		}
 	}()
-	for _, op := range fbo.dirOps {
-		md.AddOp(op.dirOp)
+
+	newBlocks := make(map[BlockPointer]bool)
+	fileBlocks := make(fileBlockMap)
+	parentsToAddChainsFor := make(map[BlockPointer]bool)
+	for _, dop := range fbo.dirOps {
+		// Copy the op before modifying it, in case there's an error
+		// and we have to retry with the original ops.  TODO: make a
+		// better way of copying a single op.
+		ops := make(opsList, 1)
+		err = kbfscodec.Update(
+			fbo.config.Codec(), &ops, opsList([]op{dop.dirOp}))
```

Comment:
Ok, I added it to the `op` interface, which means we won't need it on `opsList` since only one op is being copied at a time here.